pkg/v1/sdk/capabilities/discovery/cluster_gvr.go

```diff
@@ -12,19 +13,41 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
-// Group represents any API group that may exist on a cluster, with ability to specify versions and resource to check for GVR.
+var (
+	// ErrInvalidWithResourceMethodArgument occurs when an empty string argument is passed into the WithResource method.
+	ErrInvalidWithResourceMethodArgument = errors.New("WithResource method must have non-empty string argument; omit method to indicate any resource")
+
+	// ErrInvalidWithVersionsMethodArgument occurs when only empty string argument(s) are passed into the WithVersions method.
+	ErrInvalidWithVersionsMethodArgument = errors.New("WithVersions method must have non-empty string argument; omit method to indicate any version")
+)
+
+// Group represents any API group that may exist on a cluster, with ability to
+// specify versions and resource to check for GVR.
 func Group(queryName, group string) *QueryGVR {
 	return &QueryGVR{
 		name:  queryName,
 		group: group,
 	}
 }
 
+// nullString represents a string that may be null or not explicitly set.
+type nullString struct {
```

Comment:
I was following the NullString pattern that a standard library uses. You can see it at https://pkg.go.dev/database/sql#NullString

I know what you're proposing, using `*string` is very popular, but I think creating a new type, `NullString` is just as valid. I like the `NullString` type because it's so easy to forget to do a nil check and end up with a nil dereference panic. The NullString, not being a pointer, is less prone to panic.

However, again, your suggestion is very popular. @rajathagasthya , would you still recommend I change to using `*string`? Even though I slightly prefer `NullString` more, I also realize people will be more familiar with `*string`, so I'm fine going either way.

Reply:
It's just as valid, but perhaps more verbose than necessary? Kubernetes uses [pointers for optional things](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required) extensively, hence the suggestion. But not a blocker.

Reply:
I'm fine either way

Reply:
> It's just as valid, but perhaps more verbose than necessary? 

I agree that it's more verbose, but I'm willing to take a slight increase in verbosity to be more safe and not have to deal with nil checks every time I want to dereference the string pointer.

Since both are valid and no strong opinions either way, I'll keep the `nullString` custom type.